File: backend/internal/modules/carriers/domain/services/carrierService.go
package services

import (
	"github.com/javierjpv/edenBooks/internal/modules/carriers/application/dto"
	"github.com/javierjpv/edenBooks/internal/modules/carriers/domain/entities"
)

type CarrierService interface {
	CreateCarrier(carrier dto.CarrierRequest) error

	UpdateCarrier(id uint, a dto.CarrierRequest) error

	DeleteCarrier(id uint) error

	GetCarrierByID(id uint) (*entities.Carrier, error)

	GetFilteredCarrieres(filters map[string]string) ([]entities.Carrier, error)
}

File: frontend/src/modules/carriers/services/carrierService.ts
import { carrierApi } from "../api/carrierApi";
import { ICarrierRequest } from "../interfaces/ICarrierRequest";
import { ICarrierService } from "../interfaces/ICarrierService";

export const CarrierService: ICarrierService = {
  CreateCarrier: async (carrier: ICarrierRequest) =>carrierApi.CreateCarrier(carrier),
  GetCarriers: async () => carrierApi.GetCarriers(),
  GetCarrierById: async (id: number) => carrierApi.GetCarrierById(id),
  UpdateCarrier: async (id: number, carrier: ICarrierRequest) =>carrierApi.UpdateCarrier(id, carrier),
  DeleteCarrier: async (id: number) => carrierApi.DeleteCarrier(id),
};
